src/test/java/com/aws/greengrass/clientdevices/auth/iot/ThingRegistryTest.java

```diff
@@ -57,6 +52,7 @@ void beforeEach() throws InvalidCertificateException {
         configTopics = Topics.of(new Context(), "config", null);
         registry = new ThingRegistry(mockIotAuthClient, domainEvents, RuntimeConfiguration.from(configTopics));
         mockCertificate = CertificateFake.of("mock-certificateId");
+        mockThing.detachCertificate(mockCertificate.getCertificateId());
```

Comment:
Red flag right here. We're obviously missing tests here, but this is a biiiig no no. There is almost never a good reason to mutate object state between tests like this. In this case, `mockThing` shouldn't be a static final like it is.

You're just asking for trouble if your setup fixtures are dependent on some previous object state. At best, this can make it hard to debug test failures. At worst, it can mask legitimate failures.

Reply:
agreed, removed this. Thanks